pkgs/development/libraries/libmad/default.nix

```diff
@@ -8,7 +8,9 @@ stdenv.mkDerivation rec {
     sha256 = "bbfac3ed6bfbc2823d3775ebb931087371e142bb0e9bb1bee51a76a6e0078690";
   };
 
-  patches = [ ./001-mips_removal_h_constraint.patch ./pkgconfig.patch ];
+  # optimize.diff is taken from https://projects.archlinux.org/svntogit/packages.git/tree/trunk/optimize.diff?h=packages/libmad
+  # It fixes a build failure in Clang and probably other places
+  patches = [ ./001-mips_removal_h_constraint.patch ./pkgconfig.patch ./optimize.diff ];
```

Comment:
I definitely could, but since the patch came from Arch Linux I figured it's probably useful on Linux and for other reasons as well. Though I suppose this is a bit presumptuous, so in order to make the merging process smoother I'm happy to make the patch conditional.